packages/block-library/src/navigation-link/edit.js

```diff
@@ -332,6 +336,28 @@ export default function NavigationLinkEdit( {
 		}
 	);
 
+	const classes = classnames( 'wp-block-navigation-link__content', {
+		'wp-block-navigation-link__placeholder': ! url,
+	} );
+
+	let missingText = '';
+	switch ( type ) {
+		case 'post':
+			missingText = __( 'Select a post' );
+			break;
+		case 'page':
+			missingText = __( 'Select a page' );
+			break;
+		case 'category':
+			missingText = __( 'Select a category' );
+			break;
+		case 'tag':
+			missingText = __( 'Select a tag' );
+			break;
+		default:
+			missingText = __( 'Add a link' );
```

Comment:
I think this default works 👍  If we'd like CPT or custom Taxonomy labels later on, I think we end up needing new post type/taxonomy labels in WP

Reply:
I think we've been leaving translator comments inline so something like the following:
```jsx
/* translators: label for missing values in navigation link block */
missingText = __( 'Add a link' );
```

Reply:
I think that we could make this more extensible in the future by moving the text to `variations.js`, but there are a bunch of other bits we'd want to make extensible as well so for now I prefer this implementation.